core/src/main/scala/org/apache/spark/util/SerializableBuffer.scala

```diff
@@ -21,12 +21,16 @@ import java.io.{EOFException, IOException, ObjectInputStream, ObjectOutputStream
 import java.nio.ByteBuffer
 import java.nio.channels.Channels
 
+import com.esotericsoftware.kryo.{Kryo, KryoSerializable}
+import com.esotericsoftware.kryo.io.{Input, Output}
+
 /**
  * A wrapper around a java.nio.ByteBuffer that is serializable through Java serialization, to make
  * it easier to pass ByteBuffers in case class messages.
  */
 private[spark]
-class SerializableBuffer(@transient var buffer: ByteBuffer) extends Serializable {
+class SerializableBuffer(@transient var buffer: ByteBuffer)
+  extends Serializable with KryoSerializable {
```

Comment:
`spark.kryo.registrationRequired` -- and it is important to get this right, since registered vs. unregistered can make a large difference in the size of users' serialized classes.